File: leetcode-solutions/3371-harshad-number/solution.c
int sumOfTheDigitsOfHarshadNumber(int num) {
    int b=num,add=0;
    for(int i=1;num!=0;i++){
        int a=num%10;
        add+=a;
        num/=10;
    }
    if(b%add==0){
        return add;
    }
    else{
        return -1;
    }
}

File: leetcode-solutions/3371-harshad-number/solution.py
class Solution(object):
    def sumOfTheDigitsOfHarshadNumber(self, x):
        sum_number=0
        temp=x
        while temp>0:
            sum_number+=temp%10
            temp//=10
        if x%sum_number==0:
            return sum_number
        return -1
